### PR TITLE
feat: make cdktf construct dependencies settable

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
The smart default is a goodie to make life easier, but the main part is making it easy to override the constructs version

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/cdktf/cdktf-construct.ts [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
    * @featured
    */
   readonly cdktfVersion: string;
+
+  /**
+   * Construct version to use
+   * @default "10.0.12"
+   * @featured
+   */
+  readonly constructsVersion?: string;
```
Comment (eladb) [inline src/cdktf/cdktf-construct.ts:17]: This is actually the minimum version. Let's change this to be semver requirement so users can express a richer set:

```suggestion
  /**
   * Construct version to use
   * @default "^10.0.12"
   */
  readonly constructsVersion?: string;
```

Also, we try to reduce the set of "featured" options to a minimum. This is unlikely something someone would really want to touch.

**Review by eladb — CHANGES_REQUESTED** [review]

File: src/cdktf/cdktf-construct.ts [hunk 2]
```diff
@@ -23,14 +30,21 @@ export class ConstructLibraryCdktf extends ConstructLibrary {
...
+      if (semver.lte(options.cdktfVersion, "0.5.0")) {
+        return "3.0.0";
+      }
+
+      return "10.0.12";
     }
 
     const ver = options.cdktfVersion;
+    const constructVersion =
+      options.constructsVersion || getDefaultConstructVersion();
```
Comment (eladb) [inline src/cdktf/cdktf-construct.ts:46]:
```suggestion
      options.constructsVersion ?? getDefaultConstructVersion();
```

File: src/cdktf/cdktf-construct.ts [hunk 3]
```diff
@@ -23,14 +30,21 @@ export class ConstructLibraryCdktf extends ConstructLibrary {
     super(options);
 
     if (!options.cdktfVersion) {
-      throw new Error('Required field cdktfVersion is not specified.');
+      throw new Error("Required field cdktfVersion is not specified.");
+    }
+
+    function getDefaultConstructVersion() {
+      if (semver.lte(options.cdktfVersion, "0.5.0")) {
+        return "3.0.0";
```
Comment (eladb) [inline src/cdktf/cdktf-construct.ts:38]:
```suggestion
        return "^3.0.0";
```

File: src/cdktf/cdktf-construct.ts [hunk 4]
```diff
@@ -23,14 +30,21 @@ export class ConstructLibraryCdktf extends ConstructLibrary {
...
+      }
+
+      return "10.0.12";
```
Comment (eladb) [inline src/cdktf/cdktf-construct.ts:41]:
```suggestion
      return "^10.0.12";
```

**Review by eladb — CHANGES_REQUESTED** [review]

**Comment by DanielMSchmidt** [comment]
Thank you for your reviews and patience @eladb !

**Comment by eladb** [comment]
## Please read this to avoid merge hell

Hola! apologies for hassle. I've enabled Prettify as part of #1453. This means that most of the source files have changed and you will likely see quite a lot of merge conflicts when you try to merge from `main`.

*But there's an easy way to handle this.*

First, merge your branch from this tag:

```sh
$ git fetch origin
$ git merge 1020d0ad3f95bb4e1201f8640d440ffb43ccf230
```

Edit the `.projenrc.js` file in your branch and add this to **line 57** (right
after `codeCov: true`):

```js
  prettier: true,
```

Now, run:

```sh
$ git clean -fdx .
$ yarn
$ bash projen.bash
$ yarn eslint
$ git add .
$ git commit -m 'format code'
```

This will reformat all the code in your branch to match `main`.

Now, merge from main:

```sh
$ git merge origin/main
```

Sorry again for the inconvenience...

**Comment by DanielMSchmidt** [comment]
Closed in favor of #1464 and 🎉  for prettier 🎉

**Pull request closed without merging**